src/Compilers/CSharp/Test/Syntax/Parsing/RawInterpolatedStringLiteralParsingTests.cs

```diff
@@ -0,0 +1,1420 @@
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{""""""}""""""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void MultipleAtSigns1()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@;
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS1646: Keyword, identifier, or string expected after verbatim specifier: @
+            //         var v = @@;
+            Diagnostic(ErrorCode.ERR_ExpectedVerbatimLiteral, "@").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void MultipleAtSigns2()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = @@";
+            Diagnostic(ErrorCode.ERR_CannotMixVerbatimAndRawStrings, @"@@""").WithLocation(6, 17),
+            // (6,21): error CS1002: ; expected
+            //         var v = @@";
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(6, 21));
+    }
+
+    [Fact]
+    public void MultipleAtSigns3()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@"" "";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = @@" ";
+            Diagnostic(ErrorCode.ERR_CannotMixVerbatimAndRawStrings, @"@@""").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void MultipleAtSigns4()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@"""""" """""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = @@""" """;
+            Diagnostic(ErrorCode.ERR_CannotMixVerbatimAndRawStrings, @"@@""""""").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void MultipleAtSigns5()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@@;
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS1646: Keyword, identifier, or string expected after verbatim specifier: @
+            //         var v = @@@;
+            Diagnostic(ErrorCode.ERR_ExpectedVerbatimLiteral, "@").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void MultipleAtSigns6()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@@"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = @@@";
+            Diagnostic(ErrorCode.ERR_CannotMixVerbatimAndRawStrings, @"@@@""").WithLocation(6, 17),
```

Comment:
Diagnostic quality: the "raw string" part of the diagnostic doesn't make much sense in this scenario. Maybe we can tweak the message ("Sequence of '@' characters is not allowed. A verbatim string only needs one '@' character and a raw string doesn't allow any")?